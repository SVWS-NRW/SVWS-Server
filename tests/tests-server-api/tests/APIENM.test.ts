import {describe, expect, test} from "vitest";
import {getApiServer} from "./utils/TestUtils.js";
import {
	ArrayList, ENMFach, ENMFloskelgruppe,
	ENMFoerderschwerpunkt,
	ENMJahrgang,
	ENMKlasse, ENMLehrer,
	ENMLerngruppe,
	ENMNote,
	ENMSchueler, ENMTeilleistungsart
} from "@core";

describe("APIENM Tests", () => {
	describe.each([{schema: "GymAbi01"}])('gegen %s', ({schema}) => {
		const api = getApiServer(schema);

		test("getLehrerENMDaten", async () => {
			const result = await api.getLehrerENMDaten(schema, 76);
			expect(result).toMatchSnapshot({
				noten: expect.any(ArrayList<ENMNote>),
				foerderschwerpunkte: expect.any(ArrayList<ENMFoerderschwerpunkt>),
				jahrgaenge: expect.any(ArrayList<ENMJahrgang>),
				klassen: expect.any(ArrayList<ENMKlasse>),
				floskelgruppen: expect.any(ArrayList<ENMFloskelgruppe>),
				lehrer: expect.any(ArrayList<ENMLehrer>),
				faecher: expect.any(ArrayList<ENMFach>),
				teilleistungsarten: expect.any(ArrayList<ENMTeilleistungsart>),
				lerngruppen: expect.any(ArrayList<ENMLerngruppe>),
				schueler: expect.any(ArrayList<ENMSchueler>)
			});
			expect(result.noten.size()).toMatchSnapshot();
			expect(result.foerderschwerpunkte.size()).toMatchSnapshot();
			expect(result.jahrgaenge.size()).toMatchSnapshot();
			expect(result.klassen.size()).toMatchSnapshot();
			expect(result.floskelgruppen.size()).toMatchSnapshot();
			expect(result.lehrer.size()).toMatchSnapshot();
			expect(result.faecher.size()).toMatchSnapshot();
			expect(result.teilleistungsarten.size()).toMatchSnapshot();
			expect(result.lerngruppen.size()).toMatchSnapshot();
			expect(result.schueler.size()).toMatchSnapshot();
		});
	})
})
