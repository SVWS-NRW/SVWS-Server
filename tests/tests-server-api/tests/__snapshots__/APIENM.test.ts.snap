// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`APIENM Tests > gegen { schema: 'GymAbi01' } > getLehrerENMDaten 1`] = `
{
  "aktuellerAbschnitt": 2,
  "ankreuzkompetenzen": ENMAnkreuzkompetenzenKatalog {
    "kompetenzen": ArrayList {
      "addFirst": [Function],
      "addLast": [Function],
      "elementData": [],
      "getFirst": [Function],
      "getLast": [Function],
      "modCount": 0,
      "removeFirst": [Function],
      "removeLast": [Function],
    },
    "textSonstiges": null,
    "textStufen": [
      null,
      null,
      null,
      null,
      null,
    ],
  },
  "anzahlAbschnitte": 2,
  "enmRevision": 1,
  "faecher": Any<ArrayList>,
  "fehlstundenEingabe": true,
  "fehlstundenSIFachbezogen": false,
  "fehlstundenSIIFachbezogen": true,
  "floskelgruppen": Any<ArrayList>,
  "foerderschwerpunkte": Any<ArrayList>,
  "jahrgaenge": Any<ArrayList>,
  "klassen": Any<ArrayList>,
  "lehrer": Any<ArrayList>,
  "lehrerID": 76,
  "lerngruppen": Any<ArrayList>,
  "mailadresse": null,
  "noten": Any<ArrayList>,
  "publicKey": null,
  "schueler": Any<ArrayList>,
  "schulform": "GY",
  "schuljahr": 2018,
  "schulnummer": 168890,
  "teilleistungsarten": Any<ArrayList>,
}
`;

exports[`APIENM Tests > gegen { schema: 'GymAbi01' } > getLehrerENMDaten 2`] = `25`;

exports[`APIENM Tests > gegen { schema: 'GymAbi01' } > getLehrerENMDaten 3`] = `10`;

exports[`APIENM Tests > gegen { schema: 'GymAbi01' } > getLehrerENMDaten 4`] = `4`;

exports[`APIENM Tests > gegen { schema: 'GymAbi01' } > getLehrerENMDaten 5`] = `7`;

exports[`APIENM Tests > gegen { schema: 'GymAbi01' } > getLehrerENMDaten 6`] = `9`;

exports[`APIENM Tests > gegen { schema: 'GymAbi01' } > getLehrerENMDaten 7`] = `15`;

exports[`APIENM Tests > gegen { schema: 'GymAbi01' } > getLehrerENMDaten 8`] = `2`;

exports[`APIENM Tests > gegen { schema: 'GymAbi01' } > getLehrerENMDaten 9`] = `7`;

exports[`APIENM Tests > gegen { schema: 'GymAbi01' } > getLehrerENMDaten 10`] = `3`;

exports[`APIENM Tests > gegen { schema: 'GymAbi01' } > getLehrerENMDaten 11`] = `170`;
